internal/controller/nonadminbackup_controller_test.go

```diff
@@ -259,37 +259,39 @@ var _ = ginkgo.Describe("Test single reconciles of NonAdminBackup Reconcile func
 				gomega.Expect(err).To(gomega.HaveOccurred())
 				gomega.Expect(err.Error()).To(gomega.ContainSubstring(scenario.resultError.Error()))
 			}
-
+			nonAdminBackupAfterReconcile := &nacv1alpha1.NonAdminBackup{}
 			gomega.Expect(k8sClient.Get(
 				ctx,
 				types.NamespacedName{
-					Name:      testNonAdminBackupName,
-					Namespace: nonAdminNamespaceName,
+					Name:      nonAdminObjectName,
+					Namespace: nonAdminObjectNamespace,
 				},
-				nonAdminBackup,
+				nonAdminBackupAfterReconcile,
 			)).To(gomega.Succeed())
-
-			gomega.Expect(checkTestNonAdminBackupStatus(nonAdminBackup, scenario.ExpectedStatus, nonAdminNamespaceName, oadpNamespaceName)).To(gomega.Succeed())
-
+			gomega.Expect(checkTestNonAdminBackupStatus(nonAdminBackupAfterReconcile, scenario.nonAdminBackupExpectedStatus, oadpNamespace)).To(gomega.Succeed())
+			if scenario.uuidCreatedByReconcile {
```

Comment:
this check needs to be moved inside `checkTestNonAdminBackupStatus` function

first expect in this `if` here is not valid, no? if namespace name is big, it wont be present in nameUUID

Reply:
I don't know how to nicely do that without changing the arguments of the `checkTestNonAdminBackupStatus` function, which I do not want to do at this point. Added TODO and comment in the code.

The problem is that the structure in the `nonAdminBackupSingleReconcileScenario` can not take advantage of variables in an easy way, so we would need to pass some extra args to the `checkTestNonAdminBackupStatus`  instead of using what's in the  `nonAdminBackupSingleReconcileScenario`, bacause we are using here the `nonAdminObjectNamespace` whcih is var `nonAdminObjectNamespace = fmt.Sprintf("test-nab-reconcile-%v", counter)` assigned on every `BeforeEach`